### PR TITLE
2443 sort universities by most application received

### PR DESCRIPTION
**Pull request by brajesh-lab**
Targets #{2443}

### Description
Order of university by most application received

### Checklist:
- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: resources/views/hr/application/edit.blade.php [hunk 1]
```diff
@@ -820,9 +820,11 @@ class="w-20 h-20 rounded-circle mr-0.5" data-toggle="tooltip"
                             @endif
                             @if ($loop->last && $applicationRound->application->hasTag('need-follow-up'))
                                 <div class="d-flex justify-content-start">
+                                    @if(($application->latestApplicationRound->followUps)->count()< 3)
                                     <button type="button"
                                         class="btn btn-sm btn-info fz-14 leading-none text-white rounded-0 py-1"
                                         data-toggle="modal" data-target="#followUpModal">Follow up</button>
+                                    @endif
```
Comment (P4NK4J) [inline resources/views/hr/application/edit.blade.php:827]: Why we need these changes ?

File: Modules/HR/Resources/views/universities/index.blade.php [hunk 2]
```diff
@@ -38,7 +38,11 @@
                 <th>Contact Details</th>
                 <th>Actions</th>
             </tr>
+           @foreach($applications as $application)
+               
+          
             @foreach($universities as $university)
+            @if(($application->name)==($university->name))
             <tr>
                 <td>
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/universities/index.blade.php:47]: Indentation is off, please change at other places as well.

File: Modules/HR/Resources/views/universities/index.blade.php [hunk 3]
```diff
@@ -38,7 +38,11 @@
                 <th>Contact Details</th>
                 <th>Actions</th>
             </tr>
+           @foreach($applications as $application)
+               
+
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/universities/index.blade.php:43]: Why making it into a loop, can't we make a relationship to fetch universities corresponding to applications ? We do have a mapping of university with hr_applicant and application mapping with application

File: Modules/HR/Http/Controllers/Universities/UniversityController.php [hunk 4]
```diff
@@ -23,11 +24,26 @@ public function __construct(UniversityServiceContract $service)
...
+        ->select(
+            'hr_universities.*',
+            'hr_universities_contacts.name as st_name',
+            'hr_universities_contacts.email',
+            'hr_universities_contacts.phone',
+            'hr_universities_contacts.designation'
+        )
+        ->orderBy(DB::raw('count(hr_applicants.hr_university_id)'), 'desc')
+        ->groupBy('hr_universities.id')
+        ->get();
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Universities/UniversityController.php:40]: Can we fetch universities directly based on whereHas mapping here ?

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments, please check

File: Modules/HR/Http/Controllers/Universities/UniversityController.php (hunk 4 above)
Comment (brajesh-lab) [reply Modules/HR/Http/Controllers/Universities/UniversityController.php]: when i have studied  about whereHas mapping ,I  have seen that the performance of  laravel eloquent is slower than Query Builder

File: resources/views/hr/application/edit.blade.php (hunk 1 above)
Comment (brajesh-lab) [reply resources/views/hr/application/edit.blade.php:827]: There is no need to change  in this file . It's  attached  from privious  issue

File: Modules/HR/Resources/views/universities/index.blade.php (hunk 3 above)
Comment (brajesh-lab) [reply Modules/HR/Resources/views/universities/index.blade.php]: I have  solved

File: resources/views/hr/application/edit.blade.php (hunk 1 above)
Comment (P4NK4J) [reply resources/views/hr/application/edit.blade.php:827]: Let's remove these changes, it would stop showing the followup button then

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Responded to the comments

**Review by P4NK4J — APPROVED** [review]
Looks good.

**Pull request merged**